llvm/include/llvm/ADT/Hashing.h

```diff
@@ -541,21 +542,14 @@ struct hash_combine_recursive_helper {
   ///
   /// This function recurses through each argument, combining that argument
   /// into a single hash.
-  template <typename T, typename ...Ts>
+  template <typename... Ts>
   hash_code combine(size_t length, char *buffer_ptr, char *buffer_end,
-                    const T &arg, const Ts &...args) {
-    buffer_ptr = combine_data(length, buffer_ptr, buffer_end, get_hashable_data(arg));
+                    const Ts &...args) {
+    (combine_data(length, buffer_ptr, buffer_end, get_hashable_data(args)),
```

Comment:
Maybe be explicit about discarding the return value?
```suggestion
    ((void)combine_data(length, buffer_ptr, buffer_end, get_hashable_data(args)),
```

Reply:
Done.  Thanks!